---
title: "20190817_integrator_maturepiRNAlevels"
author: "Toni Beltran"
date: "17/08/2019"
output: html_document
---

# Analysis of total mature piRNA abundance in integrator depleted worms

Here I look at changes in mature piRNA abundance in whole worm samples upon ints-11 RNAi treatment compared to EV controls.

```{r piRNA count data}

library(ggplot2)

setwd("/Users/ab6415/Bioinformatics/Analysis/INTEGRATOR_PAPER_ANALYSES/Fig1_mature_piRNA_abundance/")

piRNA_counts_filenames <- list.files(path="./piRNA_counts/",
    pattern="*piRNAcounts")

names <-c("germnuclei_N2_EV","germnuclei_N2_ints11","germnuclei_TFIIS_EV","germnuclei_TFIIS_ints11","wholeanimal_N2_EV_72h","wholeanimal_N2_ints11_72h","wholeanimal_GTBT_TEE144_EV","wholeanimal_GTBT_TEE145_EV","wholeanimal_GTBT_TEE146_EV","wholeanimal_GTBT_TEE147_EV","wholeanimal_GTBT_TEE144_ints11","wholeanimal_GTBT_TEE145_ints11","wholeanimal_GTBT_TEE146_ints11","wholeanimal_GTBT_TEE147_ints11")
          

piRNA_counts <- lapply(piRNA_counts_filenames,function(i){
  read.csv(paste("piRNA_counts/",i,sep=""), header=FALSE,sep="\t")
})


for (df in 1:length(piRNA_counts)){
  colnames(piRNA_counts[[df]])[3] <- names[df]
  rownames(piRNA_counts[[df]])<-piRNA_counts[[df]][,"V1"]
}

DF <- piRNA_counts[[1]][,c(1,3)]
for ( .df in piRNA_counts) {
  DF <-merge(DF,.df[,c(1,3)],by.x="V1", by.y="V1", all=TRUE)
}

colnames(DF)
DF<-DF[order(DF$V1),]

nrow(DF)
rownames(DF)<-DF$V1
DF$V1<-NULL
DF[is.na(DF)]<-0
DF$germnuclei_N2_EV.x<-NULL
colnames(DF)[1]<-"germnuclei_N2_EV"

motif_dependent_piRNAs<-lapply(piRNA_counts,function(df){return(df[c(grep("^21UR-",df$V1),grep("^piRNA_type_1",df$V1)),])})

totalcounts_motif_dependent_piRNAs<-unlist(lapply(motif_dependent_piRNAs,function(df){return(sum(df[,3]))}))
totalseqs_motif_dependent_piRNAs<-unlist(lapply(motif_dependent_piRNAs,function(df){return(nrow(df))}))


```

Here I obtain size factors for library normalization by running DEseq on miRNA counts matrices obtained from the same libraries.

```{r load miRNA count data}

miRNA_counts_filenames <- list.files(path="./miRNA_counts/",
    pattern="*miRNAcounts")
miRNA_counts_filenames<-miRNA_counts_filenames[c(2,3,4,6,8,11,1,5,7,9,10,12,13,14)]

miRNA_counts <- lapply(miRNA_counts_filenames,function(i){
  read.csv(paste("miRNA_counts/",i,sep=""), header=FALSE,sep="\t")
})


for (df in 1:length(miRNA_counts)){
  colnames(miRNA_counts[[df]])[4] <- names[df]
  rownames(miRNA_counts[[df]])<-miRNA_counts[[df]][,"V1"]
}

miR_DF <- miRNA_counts[[1]][,c(1,4)]
for ( .df in miRNA_counts) {
  miR_DF <-merge(miR_DF,.df[,c(1,4)],by.x="V1", by.y="V1", all=TRUE)
}

colnames(miR_DF)
nrow(miR_DF)
rownames(miR_DF)<-miR_DF$V1
miR_DF$V1<-NULL
miR_DF[is.na(miR_DF)]<-0

miR_DF$germnuclei_N2_EV.x<-NULL
colnames(miR_DF)[1]<-"germnuclei_N2_EV"

colSums(miR_DF)


library(DESeq2)

coldata<-data.frame(condition=colnames(miR_DF)); rownames(coldata)<-colnames(miR_DF)
cds<-DESeqDataSetFromMatrix(countData = miR_DF,colData = coldata,design=~condition)
dds<-estimateSizeFactors(cds)
sizeFactors(dds)

cor(sizeFactors(dds),colSums(miR_DF))
plot(sizeFactors(dds),colSums(miR_DF))


total_miRNAs_SFs<-colSums(miR_DF)
DESeq2_miRNAs_SFs<-sizeFactors(dds)


totalcounts_motif_dependent_piRNAs/total_miRNAs_SFs
totalcounts_motif_dependent_piRNAs/DESeq2_miRNAs_SFs


```

Here I normalize piRNA counts using the size factors derived from the miRNA analysis, and use these normalized data to calculate the fold change distributions, and the change in total counts.

```{r motdep}

#motif dependent - boxplots

DF_miRnorm<-(t(t(DF)/DESeq2_miRNAs_SFs))
colnames(DF_miRnorm)
write.table(DF_miRnorm,file="mature_pis_miRnorm.txt",quote=FALSE)

ggplot(data.frame(DF_miRnorm))+
  geom_point(aes(x=log2((germnuclei_N2_EV+1)+(germnuclei_N2_ints11+1)),
                                  y=log2(germnuclei_N2_ints11+1)-log2(germnuclei_N2_EV+1)))+
  geom_hline(yintercept = 0,linetype="dashed",col="red")+
  ylab("log2(fold change ints-11 RNAi/EV)")+
  xlab("log2(mean abundance)")

ggplot(data.frame(DF_miRnorm))+
  geom_point(aes(x=log2((wholeanimal_GTBT_TEE144_EV+1)+(wholeanimal_GTBT_TEE144_ints11+1)),
                                  y=log2(wholeanimal_GTBT_TEE144_ints11+1)-log2(wholeanimal_GTBT_TEE144_EV+1)))+
  geom_hline(yintercept = 0,linetype="dashed",col="red")+
  ylab("log2(fold change ints-11 RNAi/EV)")+
  xlab("log2(mean abundance)")

ggplot(data.frame(DF_miRnorm))+
  geom_point(aes(x=log2((wholeanimal_N2_EV_72h+1)+(wholeanimal_N2_ints11_72h+1)),
                                  y=log2(wholeanimal_N2_ints11_72h+1)-log2(wholeanimal_N2_EV_72h+1)))+
  geom_hline(yintercept = 0,linetype="dashed",col="red")+
  ylab("log2(fold change ints-11 RNAi/EV)")+
  xlab("log2(mean abundance)")


log2_foldchange_meanthr<-function(df,i1,i2,lognormcount_thr){
  df<-df[which(log2((df[,i1]+df[,i2])/2+1)>lognormcount_thr),]
  return(log2(df[,i2]+1)-log2(df[,i1]+1))
}

colnames(DF_miRnorm)

DF_miRnorm_motifdep<-DF_miRnorm[c(grep("^21UR-",rownames(DF_miRnorm)),grep("^piRNA_type_1",rownames(DF_miRnorm))),]

plot_fcdistributions<-function(df,filename,logthr,title){

fc_distributions<-data.frame(fc=c(log2_foldchange_meanthr(df,7,11,logthr),log2_foldchange_meanthr(df,8,12,logthr),log2_foldchange_meanthr(df,9,13,logthr),log2_foldchange_meanthr(df,10,14,logthr),log2_foldchange_meanthr(df,5,6,logthr)),
                             exp=factor(c(rep("tee144",length(log2_foldchange_meanthr(df,7,11,logthr))),rep("tee145",length(log2_foldchange_meanthr(df,8,12,logthr))),rep("tee146",length(log2_foldchange_meanthr(df,9,13,logthr))),rep("tee147",length(log2_foldchange_meanthr(df,10,14,logthr))),rep("ww72h_ints11",length(log2_foldchange_meanthr(df,5,6,logthr)))),levels=c("ww72h_ints11","tee144","tee145","tee146","tee147")))

print(ggplot(fc_distributions,aes(y=fc,x=exp))+geom_violin(trim=FALSE,fill=alpha("lightblue",0.3))+geom_boxplot(width=0.1)+theme_classic()+geom_abline(slope=0,intercept = 0,col="red",linetype="dashed")+ylab("log2 fold change ints-11 RNAi/EV")+ggtitle(title))
ggsave(paste(filename,"_boxviolin.pdf",sep=""))

}

plot_fcdistributions(DF_miRnorm_motifdep,"mature_piRNAs_motifdep_miRnorm",2.32,"motif dependent, miRNA size factors")


#motif dependent - total

total_motif_dependent_piRNAs_normalised<-(totalcounts_motif_dependent_piRNAs/DESeq2_miRNAs_SFs)

norm_totalcounts<-data.frame(total_norm=total_motif_dependent_piRNAs_normalised,names=factor(names))
levels(norm_totalcounts$names)
norm_totalcounts$names<-factor(norm_totalcounts$names,levels(norm_totalcounts$names))
norm_totalcounts$experiment<-c(rep("germ nuclei",4),rep("whole animal E1",2),rep("whole animal GTBT",8))
norm_totalcounts$condition<-c(rep(c("EV","ints-11 RNAi"),3),rep("EV",4),rep("ints-11 RNAi",4))

ggplot(norm_totalcounts)+
  geom_col(aes(y=total_norm,x=names,fill=condition,group=experiment),position="dodge")+
  scale_fill_brewer(palette="Purples",direction=-1) +
  theme_classic()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
ggsave("normalized_total_motifdep_piRNA_content.pdf")

norm_totalcounts$total_norm_scaled_to_EV<-c(1,747733.0/987053.9,1,349437.0/899415.5,1,172292.4/367311.7,
                                            1,1,1,1,329880.3/616964.0,174226.8/425125.4,250269.1/454015.1,306492.8/487107.6)


ggplot(norm_totalcounts)+
  geom_col(aes(y=total_norm_scaled_to_EV,x=names,fill=condition,group=experiment))+
  scale_fill_brewer(palette="Purples",direction=-1)+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
ggsave("normalized_total_motifdep_piRNA_content_scaled_to_EV.pdf")


EV_mean<-mean(norm_totalcounts[which(norm_totalcounts$condition=="EV"),"total_norm_scaled_to_EV"])
ints11_mean<-mean(norm_totalcounts[which(norm_totalcounts$condition=="ints-11 RNAi"),"total_norm_scaled_to_EV"])

EV_sd<-sd(norm_totalcounts[which(norm_totalcounts$condition=="EV"),"total_norm_scaled_to_EV"])
ints11_sd<-sd(norm_totalcounts[which(norm_totalcounts$condition=="ints-11 RNAi"),"total_norm_scaled_to_EV"])

EV_se<-EV_sd/sqrt(length(norm_totalcounts[which(norm_totalcounts$condition=="EV"),"total_norm_scaled_to_EV"]))
ints11_se<-ints11_sd/sqrt(length(norm_totalcounts[which(norm_totalcounts$condition=="ints-11 RNAi"),"total_norm_scaled_to_EV"]))

df_errorbars<-data.frame(mean=c(EV_mean,ints11_mean),se=c(EV_se,ints11_se),condition=c("EV","ints-11 RNAi"))
#Error bars represent standard error of the mean
ggplot(df_errorbars, aes(x=condition, y=mean,fill=condition)) + 
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                  width=.2,                    # Width of the error bars
                  position=position_dodge(.9))+
  theme_classic()+
  ylab("total piRNA abundance relative to EV")
ggsave("EV_vs_ints11RNAi_mature_motifdep_piRNAabundance_combined_mean_SE.pdf")

```

Here I collect all mapped reads, and reads corresponding to structural RNAs (rRNA, tRNA, snRNA, snoRNA), and compute the total number of nonstructural mapped reads. These totals are then scaled and used as size factors for normalization of piRNA count data, and the same analysis as above is carried out.

```{r nonstructural size factors}

#rRNA

rRNA_counts_filenames <- list.files(path="./nonstructural_counts/",
    pattern="*rRNA")
rRNA_counts_filenames<-rRNA_counts_filenames[c(2,3,4,6,8,11,1,5,7,9,10,12,13,14)]

rRNA_counts <- lapply(rRNA_counts_filenames,function(i){
  df<-read.csv(paste("nonstructural_counts/",i,sep=""), header=FALSE,sep="\t")
  df<-df[which(df$V8=="+"),]
  return(sum(df$V6))})

rRNA_seqs <- lapply(rRNA_counts_filenames,function(i){
  df<-read.csv(paste("nonstructural_counts/",i,sep=""), header=FALSE,sep="\t")
  df<-df[which(df$V8=="+"),]
  return(nrow(df))}
  )


#tRNA

tRNA_counts_filenames <- list.files(path="./nonstructural_counts/",
    pattern="*tRNA")
tRNA_counts_filenames<-tRNA_counts_filenames[c(2,3,4,6,8,11,1,5,7,9,10,12,13,14)]

tRNA_counts <- lapply(tRNA_counts_filenames,function(i){
  df<-read.csv(paste("nonstructural_counts/",i,sep=""), header=FALSE,sep="\t")
  df<-df[which(df$V8=="+"),]
  return(sum(df$V6))})

tRNA_seqs <- lapply(tRNA_counts_filenames,function(i){
  df<-read.csv(paste("nonstructural_counts/",i,sep=""), header=FALSE,sep="\t")
  df<-df[which(df$V8=="+"),]
  return(nrow(df))}
  )


#snRNA

snRNA_counts_filenames <- list.files(path="./nonstructural_counts/",
    pattern="*snRNA")
snRNA_counts_filenames<-snRNA_counts_filenames[c(2,3,4,6,8,11,1,5,7,9,10,12,13,14)]

snRNA_counts <- lapply(snRNA_counts_filenames,function(i){
  df<-read.csv(paste("nonstructural_counts/",i,sep=""), header=FALSE,sep="\t")
  df<-df[which(df$V8=="+"),]
  return(sum(df$V6))})

snRNA_seqs <- lapply(snRNA_counts_filenames,function(i){
  df<-read.csv(paste("nonstructural_counts/",i,sep=""), header=FALSE,sep="\t")
  df<-df[which(df$V8=="+"),]
  return(nrow(df))}
  )

#snoRNA

snoRNA_counts_filenames <- list.files(path="./nonstructural_counts/",
    pattern="*snoRNA")
snoRNA_counts_filenames<-snoRNA_counts_filenames[c(2,3,4,6,8,11,1,5,7,9,10,12,13,14)]

snoRNA_counts <- lapply(snoRNA_counts_filenames,function(i){
  df<-read.csv(paste("nonstructural_counts/",i,sep=""), header=FALSE,sep="\t")
  df<-df[which(df$V8=="+"),]
  return(sum(df$V6))})

snoRNA_seqs <- lapply(snoRNA_counts_filenames,function(i){
  df<-read.csv(paste("nonstructural_counts/",i,sep=""), header=FALSE,sep="\t")
  df<-df[which(df$V8=="+"),]
  return(nrow(df))}
  )

#total mapped
total_counts_filenames <- list.files(path="./total_mapped_counts/",
    pattern="*bed")
total_counts_filenames<-total_counts_filenames[c(2,3,4,6,8,11,1,5,7,9,10,12,13,14)]

total_counts <- lapply(total_counts_filenames,function(i){
  df<-read.csv(paste("total_mapped_counts/",i,sep=""), header=FALSE,sep="\t")
  df<-df[which(df$V8=="+"),]
  return(sum(df$V6))})

total_seqs <- lapply(total_counts_filenames,function(i){
  df<-read.csv(paste("total_mapped_counts/",i,sep=""), header=FALSE,sep="\t")
  df<-df[which(df$V8=="+"),]
  return(nrow(df))}
  )

nonstructural_reads<- unlist(total_counts) - unlist(rRNA_counts) - unlist(tRNA_counts) - unlist(snRNA_counts) - unlist(snoRNA_counts)
nonstructural_seqs<- unlist(total_seqs) - unlist(rRNA_seqs) - unlist(tRNA_seqs) - unlist(snRNA_seqs) - unlist(snoRNA_seqs)

nonstructural_reads_SFs<-nonstructural_reads/(mean(nonstructural_reads))

#normalization of the counts matrix

DF_totalnorm<-t(t(DF)/nonstructural_reads_SFs)

DF_totalnorm_motifdep<-data.frame(DF_totalnorm[c(grep("^21UR-",rownames(DF_totalnorm)),grep("^piRNA_type_1",rownames(DF_totalnorm))),])

plot_fcdistributions(DF_totalnorm_motifdep,"mature_piRNAs_motifdep_totalnorm",5,"motif dependent, total mapped size factors")

ggplot(DF_totalnorm_motifdep)+geom_point(aes(x=log2((wholeanimal_N2_EV_72h+wholeanimal_N2_ints11_72h+2)/2),y=log2(wholeanimal_N2_ints11_72h)-log2(wholeanimal_N2_EV_72h+1)))+geom_abline(slope=0,intercept=0,col="red",linetype="dashed")


#total motif dependent

total_motif_dependent_piRNAs_normalised<-(totalcounts_motif_dependent_piRNAs/nonstructural_reads*1e6)
norm_totalcounts<-data.frame(total_norm=total_motif_dependent_piRNAs_normalised,
                             names=factor(c(names)))

levels(norm_totalcounts$names)
norm_totalcounts$names<-factor(norm_totalcounts$names,levels(norm_totalcounts$names))
norm_totalcounts$experiment<-c(rep("germ nuclei",4),rep("whole animal E1",2),rep("whole animal GTBT",8))
norm_totalcounts$condition<-c(rep(c("EV","ints-11 RNAi"),3),rep("EV",4),rep("ints-11 RNAi",4))


ggplot(norm_totalcounts)+
  geom_col(aes(y=total_norm,x=names,fill=condition,group=experiment),position="dodge")+
  scale_fill_brewer(palette="Purples",direction=-1) +
  theme_classic()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  ggtitle("motif dependent")
ggsave("normalized_total_piRNA_content_totalmappednorm.pdf")


norm_totalcounts$total_norm_scaled_to_EV<-c(1,104314.22/73102.76,1,35261.20/64507.94,1,39599.56/99512.42,
                                            1,1,1,1,72771.70/84953.24,35603.01/77310.03,56495.22/79318.24,68701.17/95946.07)

ggplot(norm_totalcounts)+
  geom_col(aes(y=total_norm_scaled_to_EV,x=names,fill=condition,group=experiment))+
  scale_fill_brewer(palette="Purples",direction=-1)+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  ggtitle("motif dependent")
ggsave("normalized_total_piRNA_content_scaled_to_EV_totalmappednorm.pdf")


```


```{r session info}

sessionInfo()

```
